libvast/src/system/node_control.cpp

```diff
@@ -17,11 +17,13 @@
 namespace vast::system {
 
 caf::expected<caf::actor>
-spawn_at_node(caf::scoped_actor& self, node_actor node, invocation inv) {
+spawn_at_node(caf::scoped_actor& self, const node_actor& node, invocation inv) {
   caf::expected<caf::actor> result = caf::no_error;
-  self->request(node, caf::infinite, atom::spawn_v, std::move(inv))
-    .receive([&](caf::actor actor) { result = std::move(actor); },
-             [&](caf::error err) { result = std::move(err); });
+  self
+    ->request(node, defaults::system::initial_request_timeout, atom::spawn_v,
+              std::move(inv))
+    .receive([&](caf::actor& actor) { result = std::move(actor); },
+             [&](caf::error& err) { result = std::move(err); });
```

Comment:
I'm a bit uncomfortable with moving variables out of the frame of a calling function. If CAF doesn't move the arguments into the callback there must be some reason for it. And even if they aren't accessed anymore now, that might change in the future.

Reply:
See here for why this is the correct thing to do: https://github.com/actor-framework/actor-framework/discussions/1220

Reply:
Moving variables to the parent scope is common practice for blocking actors. We've been doing this for a while, actually.